### PR TITLE
Add support for GrandPa warp sync in the node template

### PR DESCRIPTION
**Pull request by tomaka**
GrandPa warp sync is quite an important protocol nowadays, for both light clients and for https://github.com/paritytech/substrate/pull/8884.

This PR adds GrandPa warp sync in the node template, as it was missing.

### CODE REVIEW

**Review by bkchr — APPROVED** [review]
However, I think in general a tutorial would be better.

**Review by adoerr — APPROVED** [review]
Or at least add a comment, what this is used for and that it is basically optional.

**Comment by tomaka** [comment]
The node-template is basically a thousand lines of completely undocumented code that nobody understands, and you want me to write a tutorial or an explanation specifically for these specific 9 lines of code that I'm adding?

**Comment by adoerr** [comment]
> The node-template is basically a thousand lines of completely undocumented code 

Fair enough. Those thousand lines of undocumented code probably came into existence by merging one 'it's just those few lines" PR after another.

File: bin/node-template/node/src/service.rs [hunk 1]
```diff
@@ -149,6 +149,17 @@ pub fn new_full(mut config: Configuration) -> Result<TaskManager, ServiceError>
...
+		sc_finality_grandpa_warp_sync::request_response_config_for_chain(
+			&config,
+			task_manager.spawn_handle(),
+			backend.clone(),
+			grandpa_link.shared_authority_set().clone(),
+		)
+	);
+
+	let (network, network_status_sinks, system_rpc_tx, network_starter) =
+
```
Comment (tomaka) [inline bin/node-template/node/src/service.rs:162]:
```suggestion
```

**Comment by tomaka** [comment]
> Those thousand lines of undocumented code probably came into existence by merging one 'it's just those few lines" PR after another.

The code in the node-template is initially just the same code as the regular node that we have copy-pasted. It has changed very little since then, and is just kept up to date with the rest of the node from time to time.

Nobody has ever bothered to document anything because even today we're still not sure about what the API of the service stuff should look like. Everyone agrees that this code should change in a major way, and nobody documents anything because they anticipate that the API will change in a major way anyway.

**Comment by bkchr** [comment]
Needs a master merge

**Comment by arkpar** [comment]
Obsoleted by #9284 anyway

**Pull request closed without merging**